### PR TITLE
Deselect subset selection tool on deleting active subset

### PR DESCRIPTION
**Pull request by rosteen**
This ensures that the "ghost" subset outline also disappears when the subset being actively edited is deleted.

### CODE REVIEW

**Comment by rosteen** [comment]
Working on adding a test...

File: jdaviz/configs/default/plugins/viewers.py
Comment (pllim) [inline jdaviz/configs/default/plugins/viewers.py:195]: Why not use `isinstance`?

File: CHANGES.rst
Comment (pllim) [inline CHANGES.rst:79]: The change log entry is under general category but only cubeviz viewer calls your new method. I don't quite understand.

Comment (pllim) [reply CHANGES.rst:79]: Works otherwise but I am still confused why only Cubeviz needs patching.

File: jdaviz/configs/default/plugins/viewers.py
Comment (rosteen) [reply jdaviz/configs/default/plugins/viewers.py:195]: Because I was lazy and copied this from the existing `_on_subset_create`....I'll go ahead and update both.

File: CHANGES.rst
Comment (rosteen) [reply CHANGES.rst:79]: Cubeviz is the only viewer that overrides the parent class `_on_subset_delete`, so it needs the `super()` call to trigger this new code. The others get it automatically.

File: jdaviz/configs/default/plugins/viewers.py
Comment (kecnry) [reply jdaviz/configs/default/plugins/viewers.py:195]: there might be a race condition on the imports... but if so, let's add a comment

**Comment by kecnry** [comment]
I can reproduce this in glue-jupyter - should the fix be migrated upstream?  Or an issue opened there and then this can be reverted on our end once it is fixed there?

**Comment by rosteen** [comment]
> I can reproduce this in glue-jupyter - should the fix be migrated upstream? Or an issue opened there and then this can be reverted on our end once it is fixed there?

I like the latter option, I think. Lets us fix this now without needing to wait for approval/merge there, and I think it will be just slightly more complicated to do upstream.

**Comment by rosteen** [comment]
I just opened https://github.com/glue-viz/glue-jupyter/issues/401 as a reminder that we want to upstream this at some point.

**Comment by pllim** [comment]
What's the rush though? I don't think fixes in glue-astronomy take that long as long as we remember to prod Derek or Tom R about it. Maybe 2 weeks max?

**Comment by rosteen** [comment]
I added a comment linking to the follow-up issues, and changed from checking string viewer name for Mosviz Table to an `isinstance` check.

File: jdaviz/configs/default/plugins/viewers.py
Comment (kecnry) [reply jdaviz/configs/default/plugins/viewers.py]: CI is failing with the `isinstance` change, but might just need `jdaviz.configs.mosviz.plugins.viewers` (plugins is missing).

**Review by pllim — APPROVED** [review]
Works. Thanks!

File: jdaviz/configs/default/plugins/viewers.py
Comment (rosteen) [reply jdaviz/configs/default/plugins/viewers.py]: Whoops, fixed.

**Review by kecnry — APPROVED** [review]
This can be a little over-agressive and disable tools in other viewers (if you create a spatial subset in the flux-viewer, leave that tool active, then create a spectral subset in the spectrum-viewer and immediately delete that subset, both viewer's have their tools reset), but I don't think that's a deal-breaker and avoiding it would overcomplicate the logic.  LGTM - thanks!

**Pull request merged**
